### PR TITLE
fix: Check conversations affected by device changes (WEBAPP-5301)

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/entity/Conversation.js [hunk 1]
```diff
@@ -105,10 +105,14 @@ z.entity.Conversation = class Conversation {
     this.is_cleared = ko.pureComputed(() => this.last_event_timestamp() <= this.cleared_timestamp());
     this.is_muted = this.muted_state;
     this.is_verified = ko.pureComputed(() => {
-      if (this.self && (this.participating_user_ets().length || !this.participating_user_ids().length)) {
-        const all_users = [this.self].concat(this.participating_user_ets());
-        return all_users.every(user_et => user_et.is_verified());
+      const hasMappedUsers = this.participating_user_ets().length || !this.participating_user_ids().length;
```
Comment (atomrc) [inline app/script/entity/Conversation.js:108]: Quick question out of the scope of this PR, but I don't really understand this condition.

Why do we consider that there are some `mappedUsers` when either there are some user entities in the `participating_user_ets` array **or** if the `participating_user_ids` is **empty**?

File: app/script/conversation/ConversationVerificationStateHandler.js [hunk 2]
```diff
@@ -166,18 +171,31 @@ z.conversation.ConversationVerificationStateHandler = class ConversationVerifica
   }
 
   /**
-   * Get all conversation where self user is active
+   * Get all conversation where self user and the given users are active.
+   *
    * @private
+   * @param {Array<string>} userIds - Multiple user IDs (can include self user ID)
    * @returns {Array<z.entity.Conversation>} Array of conversation entities
```
Comment (atomrc) [inline app/script/conversation/ConversationVerificationStateHandler.js]: Return signature needs to be updated

File: app/script/conversation/ConversationVerificationStateHandler.js [hunk 3]
```diff
@@ -166,18 +171,31 @@ z.conversation.ConversationVerificationStateHandler = class ConversationVerifica
...
+          const selfUserId = this.conversationRepository.selfUser().id;
+          const userIdsInConversation = conversationEntity.participating_user_ids().concat(selfUserId);
+          const matchingUserIds = _.intersection(userIdsInConversation, userIds);
+
+          if (!!matchingUserIds.length) {
+            return {conversationEntity, userIds: matchingUserIds};
+          }
+        }
+      })
+      .filter(activeConversationInfo => activeConversationInfo);
```
Comment (atomrc) [inline app/script/conversation/ConversationVerificationStateHandler.js]: Let's explicitly cast it into a boolean `!!activeConversationInfo`

File: app/script/client/ClientRepository.js [hunk 4]
```diff
@@ -545,22 +540,47 @@ z.client.ClientRepository = class ClientRepository {
...
+
+  /**
+   * Update clients of a user with the given backend data.
+   *
+   * @private
+   * @param {string} userId - ID of user whose clients are updated
+   * @param {Object} clientsData - Clients data from backend
+   * @returns {Promise} Resolves when the clients have been updated
+   */
+  _updateClientsByUserId(userId, clientsData) {
```
Comment (atomrc) [inline app/script/client/ClientRepository.js]: I find it quite confusion to have a `_updateClientByUserId` and `_updateClientForUser` that both have exactly the same signature and that basically do the same stuff (except that one also `amplify.publish`).

Can't we unify those 2 (eventually add an optional param that says if the method should silent or not regarding `amplify`).
WDYT?

Comment (gregor) [reply app/script/client/ClientRepository.js]: Good point

**Pull request merged**